include/cocaine/rpc/asio/encoder.hpp

```diff
@@ -111,7 +112,23 @@ struct encoded:
         typedef typename event_traits<Event>::argument_type argument_type;
 
         type_traits<argument_type>::pack(packer, std::forward<Args>(args)...);
+        packer.pack_array(3);
+
+        auto trace_id = 0; //tracer::current_span()->get_trace_id();
+        header_t::str trace_id_data{reinterpret_cast<const char*>(&trace_id), 8};
```

Comment:
This one looks weird, I don't understand what it does.

1. Is it necessary to use `reinterpret_cast`?
2. What that magic 8 means?